src/xiana/route/helpers.clj

```diff
@@ -16,3 +23,73 @@
           (assoc :error e)
           (assoc :response
                  {:status 500 :body "Internal Server error"})))))
+
+(defonce all-methods
+  [:get :patch :trace :connect :delete :head :post :options :put])
+
+(def routes->routes'
+  #(-> (fn *** [[url opt-map & nested-routes :as route]]
+         (let [new-opt-map (if (:action opt-map)
+                             (let [action' (:action opt-map)
+                                   swagger-base-of-endpoint (:swagger-* opt-map)]
+                               (reduce (fn [acc method]
+                                         (-> acc
+                                             (assoc-in [method :handler] identity)
+                                             (assoc-in [method :action] action')
+                                             (merge swagger-base-of-endpoint)))
+                                       opt-map
+                                       all-methods))
+                             (let [swagger-base-of-endpoint (get opt-map :swagger-* {})]
+                               (reduce (fn [acc method]
+                                         (if (get acc method)
+                                           (if (get-in acc [method :handler])
+                                             acc
+                                             (-> acc
+                                                 (assoc-in [method :handler] identity)
+                                                 (merge swagger-base-of-endpoint)))
+                                           acc))
+                                       opt-map
+                                       all-methods)))]
+           (if (-> route meta :no-doc)
+             nil
+             (apply conj [url new-opt-map]
+                    (map *** nested-routes)))))
+       (keep %)
+       vec))
+
+(defn routes->swagger-data [routes' & {route-opt-map :route-opt-map}]
```

Comment:
It should be part of `xiana.swagger` ns, isn't it?

Reply:
Also, fixed.